src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -617,12 +398,12 @@ def cancel_jobs(self, job_id_list: List[str] = []):
 
         for job_id in checked_jobs["job_id_list"]:
             if (
-                job_id not in self._completed_job_states
+                not self.get_job(job_id).terminal_state
```

Comment:
Are these checks the same? Can a job be in a terminal_state and have `_check_job_terminated` be False? Or, the opposite question - can the `Job` object (the `.terminal_state` check) differ from what the `JobManager` (`_check_job_terminated`) would return?

Reply:
FYI - at some point soon-ish the `_check_job_terminated` method is going to be removed -- when cancelling a job, it first hits an ee2 endpoint that tells you whether the job has been terminated. It would be simpler just to remove that call and send a request to terminate the job. ee2 returns a 200 response if the job is no longer running, whether that be because it was successfully cancelled or because it was already finished.

Reply:
Here's a PR for that change: https://github.com/kbase/narrative/pull/2372